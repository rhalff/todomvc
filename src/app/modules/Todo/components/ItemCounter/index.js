// @flow
import React from 'react'
import styles from './scss/ItemCounter.scss'

type Props = {
  itemCount: number
}

const ItemCounter = ({ itemCount }: Props) => (
  <div className={styles.itemCounter}>
    <span>
      {itemCount} {itemCount === 1 ? 'item' : 'items'} left
    </span>
  </div>
)

export default ItemCounter
