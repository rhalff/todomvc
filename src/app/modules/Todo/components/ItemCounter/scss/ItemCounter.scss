.itemCounter {
  text-align: left;
  span {
    font-weight: 300;
  }
}
