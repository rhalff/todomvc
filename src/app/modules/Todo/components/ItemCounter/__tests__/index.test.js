// @flow
import React from 'react'
import ItemCounter from '../'
import renderer from 'react-test-renderer'

test('Can show item count', () => {
  const component = renderer.create(
    <ItemCounter
      items={[{ id: 1, text: 'Test Todo' }, { id: 2, text: 'Test Todo 2' }]}
    />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
