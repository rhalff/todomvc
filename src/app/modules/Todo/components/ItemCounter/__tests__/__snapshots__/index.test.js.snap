// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show item count 1`] = `
<div
  className="itemCounter"
>
  <span>
     
    items
     left
  </span>
</div>
`;
